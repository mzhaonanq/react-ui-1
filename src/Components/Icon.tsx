import React from 'react';
import './style/icon.scss'
import {classnames} from '../lib/helpers';
let importAll = (requireContext: __WebpackModuleApi.RequireContext) => requireContext.keys().forEach(requireContext);
try {importAll(require.context('./', true, /\.svg$/));} catch (error) {console.log(error);}

interface Props extends React.SVGAttributes<SVGElement>{
  name?: string
}
const Icon: React.FunctionComponent<Props> = (props) => {
  const {className,name,...rest}=props
  return (
    <div>
      <svg className={classnames('ui-icon',className)} {...rest}>
        <use xlinkHref={`#${name}`}/>
      </svg>
    </div>
  );
};

export {Icon}