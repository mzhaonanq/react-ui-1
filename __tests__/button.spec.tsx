import {Button} from '../src/Components/Button';
import * as renderer from 'react-test-renderer'
import * as React from 'react';
describe("number test", ()=>{
  it('1 is true', ()=>{
    expect(1).toBeTruthy()
  })
  test('0 is false',()=>{
    expect(0).toBeFalsy()
  })
  test("button is div",()=>{
    const json =renderer.create(<Button/>).toJSON()
    expect(json).toMatchSnapshot()
  })
})