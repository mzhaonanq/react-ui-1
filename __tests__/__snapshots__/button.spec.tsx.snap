// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`number test button is div 1`] = `
<div>
  按钮
</div>
`;
